File: CuboidEngine/IGame.cs
namespace CuboidEngine {
	public interface IGame {
		public string GetTitle();


		public void OnLoad();
		public void OnMouseMove( float x, float y, float dx, float dy );
		public void OnMouseButton();
		public void OnMouseScroll();
		public void OnKeyStroke( Keys key, KeyState state, bool isRepeat );
		public void OnRenderTick( double dt );
		public void OnUpdateTick( double dt );
		public void OnWindowResize( float width, float height );
	}
}

File: src/main/java/com/nonumberstudios/cuboid/IGame.java
package com.nonumberstudios.cuboid;

public interface IGame {
	String getTitle();
	void onLoad();
	void onMouseMove( float x, float y, float dx, float dy );
	void onMouseButton();
	void onMouseScroll();
	void onKeyStroke( int key, int state, boolean isRepeat );
	void onRenderTick( double dt );
	void onUpdateTick( double dt );
	void onWindowResize( float width, float height );
}
